airflow/models/log.py

```diff
@@ -55,7 +55,7 @@ def __init__(self, event, task_instance=None, owner=None, extra=None, **kwargs):
             self.task_id = task_instance.task_id
             self.execution_date = task_instance.execution_date
             self.map_index = task_instance.map_index
-            if task_instance.task:
+            if hasattr(task_instance, 'task') and task_instance.task:
                 task_owner = task_instance.task.owner
```

Comment:
`hasattr` is actually just internally `try` + `getattr` so it’s probably better to do

```python
task = getattr(task_instance, "task", None)
if task:
    task_owner = task.task_owner
```